cirq/study/visualize.py

```diff
@@ -14,14 +14,21 @@
 
 """Tool to visualize the results of a study."""
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Optional
 import numpy as np
+from cirq._compat import deprecated
 
 if TYPE_CHECKING:
     from cirq.study import result
+    import matplotlib.pyplot as plt
 
 
-def plot_state_histogram(result: 'result.Result') -> np.ndarray:
+@deprecated(
+    deadline="v0.12",
+    fix="use cirq.vis.plot_state_histogram or cirq.vis.get_state_histogram instead",
+    name="cirq.study.visualize.plot_state_histogram",
+)
+def plot_state_histogram(result: 'result.Result', ax: Optional['plt.Axis'] = None) -> np.ndarray:
```

Comment:
No need to update the args for a deprecated method - you can initialize `ax` (with this type) in the function body. If someone wants to specify `ax`, they can use the new API.

Reply:
The reason I introduced new args is because it helps with testing. If I don't explicitly pass the ax variable, it'll plot the figure by default (call `fig.show()`) which results in a warning. This further results in an error when I try to catch only the deprecation warning in the test (as we get 2 warnings instead of 1).

Reply:
removed the additional arg and modified the test.